### PR TITLE
DOC: Improve docs for jax.numpy: conjugate, conj, imag and real

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -2163,24 +2163,98 @@ def radians(x: ArrayLike, /) -> Array:
...
+    Array([2.+1.j, 3.-5.j, 7.-0.j], dtype=complex64)
+  """
   check_arraylike("conjugate", x)
   return lax.conj(x) if np.iscomplexobj(x) else lax.asarray(x)
-conj = conjugate
 
 
-@implements(np.imag)
+def conj(x: ArrayLike, /) -> Array:
+  """Alias of :func:`jax.numpy.conjugate`"""
```
Comment (jakevdp) [inline jax/_src/numpy/ufuncs.py:2196]: Please add this to the list of aliases in the docstring test of `lax_numpy_test.py`

Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py:2196]: Thanks! Added to the list of alises.

**Review by jakevdp — APPROVED** [review]

**Comment by rajasekharporeddy** [comment]
Thanks @dfm. Rebased onto the latest upstream main

**Pull request merged**